src/main/java/net/snowflake/client/jdbc/SnowflakeStatementV1.java

```diff
@@ -217,7 +216,11 @@ long executeUpdateInternal(
       updateCount = ResultUtil.calculateUpdateCount(sfResultSet);
       queryID = sfResultSet.getQueryId();
       resultSetMetadataHandler(sfResultSet);
+    } catch (SnowflakeSQLException ex) {
+      queryID = ex.getQueryId();
+      throw ex;
     } catch (SFException ex) {
+      queryID = ex.getQueryId();
```

Comment:
Will the query id always presence in response for any SQLException? If it's not avail, what will happen? e.g. do we wrap IOException in SFException and it doesn't contain the query id.

Reply:
The default in this exception is to set it to `"null"` https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/core/SFException.java#L33

in both exceptions we when the queryID is missing we can set it to `"unknown"`
at least for both exceptions when we follow the defaults - the queryId will be overriden by the value from the exception (null or "unknown") so we won't receive here incorrect queryId

@sfc-gh-igarish what do you think?

Reply:
The expectation of getQueryID, to receive a valid query id after execution of a query. So we need to test what we return if someone create a JDBC statement and not execute SQL, call getQueryID(), which ever value it returns, we should return that value in the error when it's not avail. As logically it's not executed, so no query id defined yet.

Reply:
when an exception does not have valid queryID I set it to null in the statement